volmdlr/wires.py

```diff
@@ -617,8 +467,49 @@ def from_points(cls, points: List[volmdlr.Point2D]):
             edges.append(volmdlr.edges.LineSegment2D(points[i], points[i+1]))       
         
         return cls(edges)        
-        
 
+    def linesegment_crossings(self,
+                                  linesegment: 'volmdlr.edges.LineSegment2D'):
+        """
+        Returns a list of crossings in ther form of a tuple (point,
+        primitive) of the wire primitives intersecting with the line
+        """
+        results = self.line_crossings(linesegment.to_line())
+        crossings_points = []
+        for result in results:
+            if linesegment.point_belongs(result[0]):
+                crossings_points.append(result)
+        return crossings_points
+
+    def wire_crossings(self, wire):
+        '''
+        compute crossings between two wire2d
+
+        Parameters
+        ----------
+        wire : volmdlr.wires.Wire2D
+
+        Returns
+        -------
+        crossings : List[(volmdlr.Point2D, volmdlr.Primitive2D)]
+
+        '''
+
+        crossings, crossings_points = [], []
+        for primitive in wire.primitives:
+            a = self.linesegment_crossings(primitive)
+            if len(wire.primitives) == 1:
+                return a
+            if a:
+                if crossings_points:
+                    if a[0][0] not in crossings_points:
+                        crossings.append([a[0][0], a[0][1]])
+                        crossings_points.append(a[0][0])
+                else:
+                    crossings.append([a[0][0], a[0][1]])
+                    crossings_points.append(a[0][0])
+
+        return crossings
```

Comment:
I'd suggest something like that. I don't understand the first return inside the _for_ loop. 
This has to be tested, I'm not a 100% sure about what I wrote. 
```suggestion
    def wire_crossings(self, wire):
        '''
        compute crossings between two wire2d

        Parameters
        ----------
        wire : volmdlr.wires.Wire2D

        Returns
        -------
        crossings : List[(volmdlr.Point2D, volmdlr.Primitive2D)]
        '''
        crossings, crossings_points = [], []
        for primitive in wire.primitives:
            a = self.linesegment_crossings(primitive)
            if a:
                if a[0][0] not in crossings_points:
                    crossings.append([a[0][0], a[0][1]])
                    crossings_points.append(a[0][0])
        return crossings
```

Reply:
The first return inside the for loop, is when the primitive that define the wire crosses the contour twice in different places

![image](https://user-images.githubusercontent.com/68638483/158819005-48e76bf0-7055-4277-a73d-5752c04c3f94.png)

If we use your version, we'll have just one crossing point !